include/iceflow/iceflow.hpp

```
@@ -0,0 +1,154 @@
+          const std::vector<int> &nTopic, ndn::Face &face,
+          std::optional<int> publishInterval)
+      : m_face(face), m_scheduler(face.getIoContext()),
+        m_nodePrefix(nodePrefix), m_pubTopic(pubTopic), m_subTopic(subTopic),
+        m_topicPartitions(nTopic), m_publishInterval(publishInterval) {
+
+    ndn::svs::SecurityOptions secOpts(m_keyChain);
+
+    ndn::svs::SVSPubSubOptions opts;
+
+    m_svsPubSub = std::make_shared<ndn::svs::SVSPubSub>(
+        ndn::Name(syncPrefix), ndn::Name(nodePrefix), m_face,
+        std::bind(&IceFlow::fetchStateVector, this, _1), opts, secOpts);
+
+    if (!subTopic.has_value()) {
+      return;
+    }
+
+    for (auto topicPartion : m_topicPartitions) {
+      auto subscribedTopic =
+          ndn::Name(subTopic.value()).appendNumber(topicPartion);
+
+      m_svsPubSub->subscribe(
+          subscribedTopic,
+          std::bind(&IceFlow::subscribeCallBack, this, std::placeholders::_1));
+      std::cout << "Subscribed to " << subscribedTopic << std::endl;
+    }
+  }
+
+  virtual ~IceFlow() = default;
+
+public:
+  void run() {
+    std::thread svsThread([this] { m_face.processEvents(); });
+
+    if (m_pubTopic.has_value()) {
+      publishMsg();
+    }
+
+    svsThread.join();
+  }
+
+  void pushData(std::vector<uint8_t> &data) { m_outputQueue.push(data); }
+
+  std::vector<uint8_t> receiveData() { return m_inputQueue.waitAndPopValue(); }
+
+private:
+  void subscribeCallBack(const ndn::svs::SVSPubSub::SubscriptionData &subData) {
+    NDN_LOG_DEBUG("Producer Prefix: " << subData.producerPrefix << " ["
+                                      << subData.seqNo << "] : " << subData.name
+                                      << " : ");
+
+    auto data = subData.data;
+    m_inputQueue.push(std::vector<uint8_t>(data.begin(), data.end()));
+  }
+
+  void
+  fetchStateVector(const std::vector<ndn::svs::MissingDataInfo> &missing_data) {
+    // TODO: Implement if needed
```

Maybe throw an explanatory exception here to make sure it is unused?

---

Do you think something like this could work for now instead? (As throwing an exception causes the execution of the examples to fail immediately, it seems.)

```suggestion
    NDN_LOG_DEBUG("Invoking onMissingData callback");
    // TODO: Implement if needed
```